libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1094,7 +1094,7 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 			if (!boost::algorithm::all_of_equal(_statement.declarations(), nullptr))
 				m_errorReporter.fatalTypeError(_statement.location(), "Use of the \"var\" keyword is disallowed.");
 			else
-				return false;
+				solAssert(false, "Failure reported via syntax checker already");
```

Comment:
Ah sorry, I should have been more specific: Please check for `m_errorReporter.hasErrors()` - then we might change it in the future so that we can continue even after the syntax checker returned an error.

Reply:
@chriseth I actually dived into this code again, and realised that it really has to simply return there.

Also, it doesn't matter whether `true` or `false` is returned (AFAICT), because this else-branch basically means that the LHS of the var decl is not having any named components, regardless of what the RHS is. the RHS doesn't need to be inferred here either, since there's nothing in the LHS to assign those values to (and the fact that only very broken code would look like this, it's fine to bail out here straight).

I've also extended the other test files a bit to ensure that the usage of variables that couldn't be inferred aren't crashing the compiler (which they won't, because we again agreed on reporting *fatal* error, resulting in a direct bailout.

Reply:
Better be safe than sorry - we might change the AST at some point in the future where this argumentation turns out to be incorrect. If we have an assertion here, it will be noticed.